cluster-autoscaler/core/scale_down.go

```diff
@@ -608,6 +612,7 @@ func (sd *ScaleDown) UpdateUnneededNodes(
 	sd.nodeUtilizationMap = utilizationMap
 	sd.clusterStateRegistry.UpdateScaleDownCandidates(sd.unneededNodesList, timestamp)
 	metrics.UpdateUnneededNodesCount(len(sd.unneededNodesList))
+	metrics.UpdateUnremovableNodesCount(sd.unremovableReasonCounters)
```

Comment:
I guess you may need to lock the mutex here.

Reply:
I think you are right, I'll try to get to this tomorrow.

Reply:
Done.